src/directives/vHide.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Custom Vue directive v-hide
+ *
+ * Usage: v-hide="boolean"
+ *
+ * Toggles the 'visibility: hidden' style on an element.
+ * The element remains in the DOM and occupies its space, preventing layout shifts.
+ *
+ * @param {HTMLElement} el - The element the directive is bound to.
+ * @param {object} binding - The binding object, binding.value contains the directive's value.
+ */
+const setVisibility = (el, binding) => {
```

Comment:
This is an interesting way of doing this I haven't seen before. Usually I've just set a class re-actively that would have a style set with the visibility on/off, but I guess this breaks it out more explicitly.